codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/Rust.kt

```diff
@@ -109,11 +110,13 @@ object TestWorkspace {
                 PANIC("")
             }
         }
-        return TestWriterDelegator(
+        val writer = TestWriterDelegator(
             FileManifest.create(subprojectDir.toPath()),
             symbolProvider,
             CoreCodegenConfig(debugMode = debugMode),
         )
+        writer.lib { rust("// touch lib.rs") }
```

Comment:
all tests need to eventually have a lib.rs otherwise the resulting crate is broken.

If the test _fails_ before the crate is finalized, we don't end up generating one which causes all subsequent tests to fail. This prevents that failure mode